src/nvim/api/deprecated.c

```diff
@@ -522,3 +523,222 @@ Dictionary nvim_get_option_info(String name, Error *err)
+}
+
+/// Gets a window option value
+///
+/// @deprecated
+/// @param window   Window handle, or 0 for current window
+/// @param name     Option name
+/// @param[out] err Error details, if any
+/// @return Option value
+Object nvim_win_get_option(Window window, String name, Arena *arena, Error *err)
+  FUNC_API_SINCE(1)
+{
+  win_T *win = find_window_by_handle(window, err);
+
+  if (!win) {
+    return (Object)OBJECT_INIT;
+  }
+
+  return get_option_from(win, SREQ_WIN, name, err);
+}
+
+/// Sets a window option value. Passing `nil` as value deletes the option (only
+/// works if there's a global fallback)
+///
+/// @deprecated
+/// @param channel_id
+/// @param window   Window handle, or 0 for current window
+/// @param name     Option name
+/// @param value    Option value
+/// @param[out] err Error details, if any
+void nvim_win_set_option(uint64_t channel_id, Window window, String name, Object value, Error *err)
+  FUNC_API_SINCE(1)
+{
+  win_T *win = find_window_by_handle(window, err);
+
+  if (!win) {
+    return;
+  }
+
+  set_option_to(channel_id, win, SREQ_WIN, name, value, err);
+}
+
+/// Gets the value of a global or local (buffer, window) option.
+///
+/// @param from If `type` is `SREQ_WIN` or `SREQ_BUF`, this must be a pointer
+///        to the window or buffer.
+/// @param type One of `SREQ_GLOBAL`, `SREQ_WIN` or `SREQ_BUF`
+/// @param name The option name
+/// @param[out] err Details of an error that may have occurred
+/// @return the option value
+static Object get_option_from(void *from, int type, String name, Error *err)
+{
+  Object rv = OBJECT_INIT;
+
+  VALIDATE_S(name.size > 0, "option name", "<empty>", {
+    return rv;
+  });
+
+  // Return values
+  int64_t numval;
+  char *stringval = NULL;
+
+  int flags = get_option_value_strict(name.data, &numval, &stringval, type, from);
+  VALIDATE_S(flags != 0, "option name", name.data, {
+    return rv;
+  });
+
+  if (flags & SOPT_BOOL) {
+    rv.type = kObjectTypeBoolean;
+    rv.data.boolean = numval ? true : false;
+  } else if (flags & SOPT_NUM) {
+    rv.type = kObjectTypeInteger;
+    rv.data.integer = numval;
+  } else if (flags & SOPT_STRING) {
+    if (!stringval) {
+      api_set_error(err, kErrorTypeException, "Failed to get option '%s'", name.data);
+      return rv;
+    }
+    rv.type = kObjectTypeString;
+    rv.data.string.data = stringval;
+    rv.data.string.size = strlen(stringval);
+  } else {
+    api_set_error(err, kErrorTypeException, "Unknown type for option '%s'", name.data);
+  }
+
+  return rv;
+}
+
+/// Sets the value of a global or local (buffer, window) option.
+///
+/// @param to If `type` is `SREQ_WIN` or `SREQ_BUF`, this must be a pointer
+///        to the window or buffer.
+/// @param type One of `SREQ_GLOBAL`, `SREQ_WIN` or `SREQ_BUF`
+/// @param name The option name
+/// @param[out] err Details of an error that may have occurred
+void set_option_to(uint64_t channel_id, void *to, int type, String name, Object value, Error *err)
```

Comment:
If this is needed for non-deprecated API, then it shouldn't be in deprecated.c, otherwise it should be static.

Reply:
Made it static